CocktailList.js

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect } from "react";
+import { useRef, useState } from "react";
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchSearchCocktail } from "../actions/action"; 
+import { Link } from "react-router-dom";
+import { fetchCocktails } from "../actions/action"
+
+
+
+function CocktailList() {
+    const searchValue = useRef();
+    const { cocktails, loading } = useSelector((state) => ({ ...state.app }))
+    const [modifiedCocktail, setmodifiedCocktail] = useState([]);
+    const dispatch = useDispatch();
+
+
+    useEffect(() => {
+        dispatch(fetchCocktails());
+    }, []);
+
+    useEffect(() => {
+        if (cocktails) {
+            const newCocktails =  cocktails.map((item) => {
+                const { idDrink, strDrink, strDrinkThumb, strAlcoholic, strGlass } = item;
+                return {
+                    id: idDrink,
+                    name: strDrink,
+                    image: strDrinkThumb,
+                    info: strAlcoholic,
+                    glass: strGlass,
+                };
+            });
+            setmodifiedCocktail(newCocktails)
+        } else {
+            setmodifiedCocktail([]);
+        }
+    }, [cocktails]);
+    if (loading) {

```

Comment:
All renderings of a component should go to the same place, you should not have them separated.  Try using ternary operator in the main return